vertical-pod-autoscaler/pkg/recommender/model/aggregate_container_state.go

```diff
@@ -46,6 +47,10 @@ import (
 	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/util"
 )
 
+var (
+	globalPruningGracePeriodDuration = flag.Duration("pruning-grace-period-duration", 24*365*100*time.Hour, `The grace period for deleting stale aggregates and recommendations. By default, set to 100 years, effectively disabling it.`)
```

Comment:
I'm unsure what others think, but I'm not super excited by having the default be 100 years. My preference would be that setting the global to zero would disable the pruning feature (except for VPAs with pruningGracePeriod set)

Reply:
Yeah, I agree it's not ideal. I was trying to figure out how I could enable/disable it with a single flag, and without specially handling a zero value for a duration. I guess I could switch to `flag.String` and specially handle a `0` as off. Or if there's some cleaner way to do it I'm happy for feedback.

Reply:
For what it's worth, I found another flag that uses `<= 0` as a way to disable it: https://github.com/kubernetes/autoscaler/blob/ce01f0210f2b0d50f05395c7968752f548c8a75c/vertical-pod-autoscaler/pkg/updater/main.go#L62-L64